include/highfive/bits/H5_definitions.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#if defined(__GNUC__) || defined(__clang__)
+#define H5_DEPRECATED(msg) __attribute__((deprecated(#msg)))
+#elif defined(_MSC_VER)
+#define H5_DEPRECATED(msg) __declspec(deprecated(#msg))
+#else
+#pragma message("WARNING: Compiler doesnt support deprecation")
+#define H5_DEPRECATED
+#endif
+
+
+// Fw declarations
```

Comment:
```suggestion
// Forward declarations
```
It's not worth saving those few characters here ;)